src/sql/azdata.proposed.d.ts

```diff
@@ -425,6 +425,20 @@ declare module 'azdata' {
 			 */
 			None = 'None'
 		}
+
+		/*
+		 * Connection option visibility
+		*/
+		export enum OptionVisibility {
```

Comment:
I'd remove all these from the API typings - it's not something that the extensions will actually use since it all lives in the package.json. 

JSON schema, [validation](https://github.com/microsoft/azuredatastudio/issues/21023) and documentation would be how package.json contributions are presented/validated. Modifying the JSON schema isn't something we can do right now (last I checked at least) unfortunately, so it would be an excellent idea to add a sample to https://github.com/microsoft/azuredatastudio/tree/main/samples like VS Code has with [their samples](https://github.com/microsoft/vscode-extension-samples) in addition to documenting how to use this feature.

Reply:
I don't mind having them either, we can ofcourse add samples, but I don't reference them, I'm not sure if that's just me. I would prefer to have schema info in the code right there.

Reply:
This typings file is only for extension APIs, we should not be putting anything else in there.

You can (and should) define the types elsewhere like you do already for the OptionsVisibility enum in src/sql/platform/connection/common/constants.ts